src/devices/mindy.ts

```diff
@@ -0,0 +1,366 @@
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.locationDesc) {
+                const data = msg.data;
+                const wifi = data.locationDesc && data.locationDesc.trim().length > 0 ? "ON" : "OFF";
+
+                return {ip_address: data.locationDesc.trim(), wifi: wifi};
+            }
+        },
+    } satisfies Fz.Converter,
+};
+
+const optionsLocal = {
+    last_boot_update: () => {
+        return e
+            .numeric("last_boot_update", ea.STATE_SET)
+            .withValueMin(10)
+            .withDescription("Interval for request boot datetime from device. (default 60 seconds)");
+    },
+};
+
+export const definitions: DefinitionWithExtend[] = [
+    {
+        zigbeeModel: ["Leleka"],
+        model: "Leleka",
+        vendor: "MindY",
+        description: "Advanced Environmental Monitoring Device",
+        ota: true,
+        fromZigbee: [fzLocal.last_boot, fzLocal.wifi],
+        toZigbee: [tzLocal.last_boot, tzLocal.wifi],
+        options: [optionsLocal.last_boot_update()],
+        onEvent: (type, data, device, options) => {
+            const endpoint = device.getEndpoint(1);
+            if (type === "stop") {
+                clearInterval(globalStore.getValue(device, "interval"));
+                globalStore.clearValue(device, "interval");
+            } else if (!globalStore.hasValue(device, "interval")) {
+                const seconds = options?.last_boot_update ? options.last_boot_update : 60;
+                utils.assertNumber(seconds);
+                if (seconds === -1) return;
+                const interval = setInterval(async () => {
+                    try {
+                        await endpoint.read("genTime", ["lastSetTime"]);
+                        await endpoint.read("genBasic", ["locationDesc"]);
+                    } catch (error) {
+                        /* Do nothing */
+                    }
+                }, seconds * 1000);
+                globalStore.putValue(device, "interval", interval);
+            }
+        },
+
+        configure: async (device, coordinatorEndpoint) => {
+            const endpoint = device.getEndpoint(1);
+
+            await endpoint.read("genTime", ["lastSetTime"]);
+
+            await endpoint.read("genBasic", ["locationDesc"]);
+
+            await endpoint.read("genBasic", [0x0200]);
+            await endpoint.read("genBasic", [0x0240]);
+            await endpoint.read("genBasic", [0x0241]);
+            await endpoint.read("genBasic", [0x0242]);
+
+            await endpoint.read("msCO2", [0x0220]);
+            await endpoint.read("msCO2", [0x0221]);
+            await endpoint.read("msCO2", [0x0222]);
+            await endpoint.read("msCO2", [0x0223]);
+
+            await endpoint.read("msCO2", [0x0230]);
+            await endpoint.read("msCO2", [0x0231]);
+            await endpoint.read("msCO2", [0x0232]);
+
+            await endpoint.read("msIlluminanceMeasurement", [0x0220]);
```

Comment:
Are these needed? The extend below should already do this.

Reply:
without forced reading, these values remain undefined after a successful interview and device configuration. until the user requests them manually.

Reply:
Maybe I should put all custom attribute numbers in a separate object at the beginning of the file?

Reply:
Yes, and use a for loop to prevent a lot of code here.

Reply:
Yeah, it's really better that way.